packages/babel-helpers/src/helpers/AsyncGenerator.js

```diff
@@ -0,0 +1,112 @@
+/* @minVersion 7.0.0-beta.0 */
+
+import OverloadYield from "OverloadYield";
+
+export default function AsyncGenerator(gen) {
+  var front, back;
+
+  function send(key, arg) {
+    return new Promise(function (resolve, reject) {
+      var request = {
+        key: key,
+        arg: arg,
+        resolve: resolve,
+        reject: reject,
+        next: null,
+      };
+
+      if (back) {
+        back = back.next = request;
+      } else {
+        front = back = request;
+        resume(key, arg);
+      }
+    });
+  }
+
+  function resume(key, arg) {
+    try {
+      var result = gen[key](arg);
+      var value = result.value;
+      var overloaded = value instanceof OverloadYield;
```

Comment:
The brand check may fail if the generator results are produced by another `@babel/runtime` copy.

For example, a library shipped with `OverloadYield` from `@babel/runtime@7.19.0` will not work with `AsyncGenerator` from `@babel/runtime@7.19.1` because `OverloadYield` are defined in different files.

The previous helper `AwaitValue` suffers from the same issue. Since we don't have such reports yet, I don't think it is an urgent issue, we can address that in later PRs.

Reply:
`OverloadYield` is only used by other helpers and not by transpiled code, so it's always from the "correct" copy. I wish we could mark it as "internal only".

Reply:
Yes it is an internal helper, but isn't it likely that there are two copies of `_wrapAsyncGenerator` running? Say if library `foo` ships with `@babel/runtime` version A
```js
// src.js
export default async function *gen() {}
```
which is transpiled to
```js
// lib.js
function _wrapAsyncGenerator() { ... }
module.exports = _wrapAsyncGenerator( ... )
```

and an app ships with `@babel/runtime` version B
```js
// src.js
import gen from "foo";
export default async function *bar() {
  yield* foo();
}
```
which is transpiled to
```js
// lib.js
function _wrapAsyncGenerator() { ... }
module.exports = _wrapAsyncGenerator( ... )
```

The app's async generator will be wrapped by babel/runtime B while the lib's async generator will be wrapped by babel/runtime A.

Reply:
I see, the value was wrapped by the client's AwaitValue, so the copy is correct.